heatmap/src/heatmap.rs

```diff
@@ -201,20 +206,37 @@ impl Heatmap {
     /// values.
     fn tick(&self, time: Instant) {
         loop {
+            // quick check to see if this data point requires ticking forward
             let next_tick = self.next_tick.load(Ordering::Relaxed);
             if time < next_tick {
                 return;
             } else {
```

Comment:
More of a stylistic thing but you don't need the else statement here. Avoiding the else pyramid of indentation when you get nested cases is an extra win.

I'm pretty sure clippy will actually emit a warning here because you have
```
if x { ... }
else {
  if y { ... }
}
```
although not for the same reasons I am :)

Reply:
I don't think it will there, because the `if` is an `if let`. But yes, does not need to be in an `else` block